ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/recovery/RecoveryManager.java

```diff
@@ -194,22 +194,25 @@ public void waitForTermination() {
      * Suspend the recovery manager. If the recovery manager is in the process of
      * doing recovery scans then it will be suspended afterwards, in order to
      * preserve data integrity.
-     * <p>
-     * Note that this method is also influenced by
-     * {@link com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean#setWaitForWorkLeftToDo}.
-     * For more details, see {@link PeriodicRecovery#suspendScan(boolean)}
      *
      * @param async false means wait for the recovery manager to finish any scans before returning.
+     * @param waitForWorkLeftToDo when true, it is important that, before invoking this method,
+     * all transactions will either be terminated by the Transaction Reaper or they have prepared
+     * and a log has been written, otherwise the suspend call may never return.
      * @throws IllegalStateException if the recovery manager has been shutdown.
      */
+    public void suspend(boolean async, boolean waitForWorkLeftToDo) {
+        trySuspend(async, waitForWorkLeftToDo);
+    }
+
     public void suspend(boolean async) {
-        trySuspend(async);
+        trySuspend(async, false);
     }
 
-    public RecoveryManagerStatus trySuspend(boolean async) {
+    public RecoveryManagerStatus trySuspend(boolean async, boolean waitForWorkLeftToDo) {
```

Comment:
If we proceed with this, it's theoretically possible we could emit the warn_feature_not_supported_across_all_recovery_modules at this point.

_However_ I am not convinced that this is good for the user to only learn this information at this point. I wonder if we might keep the need to setWaitForWorkLeftToDo and so it can keep emitting the warning and also if someone calls trySuspend with true for the second parameter when waitForWorkLeftToDo is false we could either log an error or throw an exception.

Reply:
Please, refer to [this](https://github.com/jbosstm/narayana/pull/2359#discussion_r1979230264) comment. Thanks

Reply:
Having it documented is a great approach to give advance knowledge but I do also think that we should emit the warning at this point (to catch a situation that a user was not already aware) if someone has called this with true with a recovery module configured that we know is not compatible with JBTM-3893.

Reply:
We may have had a similar discussion to that before but I am thinking this is more an implementation detail rather than the public RecoveryModule API if we have a temporary internal marker interface we remove when they are all implemented.